src/EnergyPlus/InputProcessing/IdfParser.cc

```diff
@@ -532,13 +535,70 @@ json IdfParser::parse_number(std::string_view idf, size_t &index)
     return convert_int(value);
 }
 
+json IdfParser::parse_integer(std::string_view idf, size_t &index)
+{
+    eat_whitespace(idf, index);
+
+    size_t save_i = index;
+
+    bool running = true;
+    while (running) {
+        if (save_i == idf_size) {
+            break;
+        }
+
+        char const c = idf[save_i];
+        switch (c) {
+        case '!':
+        case ',':
+        case ';':
+        case '\r':
+        case '\n':
+            running = false;
+            break;
+        default:
+            ++save_i;
+        }
+    }
+
+    auto diff = save_i - index;
+    auto string_value = idf.substr(index, diff);
+    index_into_cur_line += diff;
+    index = save_i;
+
+    auto const string_end = string_value.data() + string_value.size(); // have to do this for MSVC
+    int int_value;
+    // Try using from_chars
+    auto result = FromChars::from_chars(string_value.data(), string_value.data() + string_value.size(), int_value);
+    if (result.ec == std::errc::result_out_of_range || result.ec == std::errc::invalid_argument) {
+        // Failure, return the string
+        return rtrim(string_value);
+    } else if (result.ptr != string_end) {
+        // Didn't use the entire string, try again via double conversion + rounding
+        size_t plus_sign = 0;
+        if (string_value.front() == '+') {
+            plus_sign = 1;
+        }
+        double double_value;
+        auto fresult = fast_float::from_chars(string_value.data() + plus_sign, string_value.data() + string_value.size(), double_value);
+        if (fresult.ec == std::errc::invalid_argument || fresult.ec == std::errc::result_out_of_range) {
+            // Failure, return the string
+            return rtrim(string_value);
+        }
+        int_value = static_cast<int>(std::round(double_value));
```

Comment:
This is where double input from IDF gets rounded to an integer.

Reply:
It feels like there should be a readily available solution to parse the integer out, but such is life I suppose.  I didn't grok every line here, but I have a feeling that if it wasn't parsing the integers properly, we'd be seeing issues.